### PR TITLE
Improving mnist example model

### PR DESCRIPTION
**Pull request by fshabashev**
I look on Keras examples and found that they have a better performing model for Mnist. 
https://github.com/keras-team/keras/blob/master/examples/mnist_cnn.py

I reimplemented their approach in PyTorch and got a better performing model.

I compared the new model with the current pytorch example.
Here are the test results:
```
Pytorch original mnist example:
Test set: Average loss: 0.0315, Accuracy: 9895/10000 (99%)
Test set: Average loss: 0.0310, Accuracy: 9894/10000 (99%)
Test set: Average loss: 0.0315, Accuracy: 9896/10000 (99%)
Test set: Average loss: 0.0317, Accuracy: 9891/10000 (99%)
Test set: Average loss: 0.0314, Accuracy: 9894/10000 (99%)
Test set: Average loss: 0.0324, Accuracy: 9894/10000 (99%)
Test set: Average loss: 0.0317, Accuracy: 9892/10000 (99%)
Test set: Average loss: 0.0314, Accuracy: 9895/10000 (99%)
Test set: Average loss: 0.0310, Accuracy: 9895/10000 (99%)
Test set: Average loss: 0.0315, Accuracy: 9893/10000 (99%)
Test set: Average loss: 0.0319, Accuracy: 9894/10000 (99%)

Pytorch proposed model example:
Test set: Average loss: 0.0261, Accuracy: 9924/10000 (99%)
Test set: Average loss: 0.0276, Accuracy: 9914/10000 (99%)
Test set: Average loss: 0.0279, Accuracy: 9921/10000 (99%)
Test set: Average loss: 0.0276, Accuracy: 9912/10000 (99%)
Test set: Average loss: 0.0279, Accuracy: 9918/10000 (99%)
Test set: Average loss: 0.0272, Accuracy: 9920/10000 (99%)
Test set: Average loss: 0.0273, Accuracy: 9917/10000 (99%)
Test set: Average loss: 0.0270, Accuracy: 9918/10000 (99%)
Test set: Average loss: 0.0264, Accuracy: 9919/10000 (99%)
Test set: Average loss: 0.0275, Accuracy: 9919/10000 (99%)
```
As you can see, accuracy for new model is higher, while logloss is lower.
I'm suggest to replace the current version with the better performing model. 
The new model is not much more complex and training time is somewhat similar. 
What do you think?

### CODE REVIEW

**Comment by fshabashev** [comment]
thank you.
apparently I don't see your comments on the diff's page, can you please send me the direct link where I can look at the changes you requested.

**Comment by fshabashev** [comment]
Cool, I've done the changes you requested.

**Comment by fshabashev** [comment]
yes, I didn't notice it.
removed these attributes after your comment

**Comment by soumith** [comment]
thank you!

**Comment by mariomeissner** [comment]
Why is there no non-linearity (`F.relu(x)`) after the second convolution (`self.conv2(x)`)?

**Comment by soumith** [comment]
@mariomeissner was a bug. fixed via https://github.com/pytorch/examples/commit/dce9319f73b13cb8d5df27e21934efce327c7eaa
